dds/DCPS/transport/framework/TransportSendBuffer.cpp

```diff
@@ -295,6 +295,10 @@ SingleSendBuffer::resend(const SequenceRange& range, DisjointSequence* gaps)
 bool
 SingleSendBuffer::resend_i(const SequenceRange& range, DisjointSequence* gaps)
 {
+  //Special case, nak to make sure it has all history
+  SequenceNumber lowForAllResent;
+  range.first == SequenceNumber() ? lowForAllResent = low() : lowForAllResent = range.first;
```

Comment:
This seems overly complex... how about
const SequenceNumber lowForAllResent = range.first == SequenceNumber() ? low() : range.first;